Software/bsw/static/Mcal/CAN/src/Can.c

```diff
@@ -1343,163 +1344,71 @@ Std_ReturnType Can_SetControllerMode( uint8 Controller, Can_ControllerStateType
                 if(DisableCnt[0U]==0U)
                 {
                     /* Enables individual CAN controller interrupt sources */
-                    HWREG(CAN0_BASE + CAN_O_CTL) |= CAN_INT_MASTER | CAN_INT_ERROR | CAN_INT_STATUS;
+                    HWREG(ui32BaseAddress + CAN_O_CTL) |= CAN_INT_MASTER | CAN_INT_ERROR | CAN_INT_STATUS;
                     /* Enable the general interrupt.*/
-                    HWREG(g_pui32EnRegs[(INT_CAN0 - 16) / 32]) =(1 << ((INT_CAN0 - 16) & 31));
+                    CANIntEnable(ui32BaseAddress,CAN_CTL_EIE | CAN_CTL_SIE | CAN_CTL_IE);
+                    //HWREG(g_pui32EnRegs[(INT_CAN0 - 16) / 32]) =(1 << ((INT_CAN0 - 16) & 31));
                     /* Enables the CAN controller for message processing.  Once enabled, the
                   controller automatically transmits any pending frames, and processes any
                   received frames.*/
```

Comment:
Why do we enable interrupt in Can_SetControllerMode and not in Can_EnableControllerInterrupts  ?

Reply:
Please have a look on the latest changes on DevCAN, I have updated the implementation